doc/specs/#653 - Quake Mode/#653 - Quake Mode.md

```diff
@@ -0,0 +1,661 @@
+      - When `dropdownDuration` is not `0`, then the window will slide back off
+        the top at the same speed as it would come down.
+  - When `false`: When this hotkey is pressed, and the terminal window is
+    currently active, do nothing.
+
+### Quake Mode
+
+In addition to just summoning the window from anywhere, some terminals also
+support a special "quake mode" buffer or window. This window is one that closely
+emulates the console from quake:
+* It's docked to the top of the screen
+* It takes the full width of the monitor, and only the bottom can be resized
+* It often doesn't have any other UI elements, like tabs
+
+For fun, we'll also be adding a special `"_quake"` window with the same
+behavior. If the user names a window `_quake`, then it will behave in the
+following special ways:
+
+* On launch, it will ignore the `initialPosition` and
+  `initialRows`/`initialCols` setting, and instead resize to the top half of the
+  monitor.
+* On launch, it will ignore the `launchMode` setting, and always launch in focus
+  mode.
+  - Users can disable focus mode on the `_quake` window if they do want tabs.
+* It will not be resizable from any side except the bottom of the window, nor
+  will it be drag-able.
+* It will not be a valid target for the "most recent window" for window
+  glomming. If it's the only open window, with `"windowingBehavior":
+  "useExisting*"`, then a new window will be created instead.
+  - It _is_ still a valid target for something like `wt -w _quake new-tab`
+
+A window at runtime can be renamed to become the `_quake` window (if no other
+`_quake` window exists). When it does, it will resize to the position of the
+quake window, and enter focus mode.
+
+We'll also be adding a special action `quakeMode`. This action is a special case
+of the `globalSummon` action, to specifically invoke the quake window in the
+current place. It is basically the same thing as the more elaborate:
+
+```json
+{
+    "monitor": "toCurrent",
+    "desktop": "toCurrent",
+    "window": "_quake",
+    "dropdownDuration": 0.5
+},
+```
+
+### Minimize to Tray
+
+Many users have requested that the terminal additionally supports minimizing the
+window "to the tray icon". This is a bit like when you close the Teams window,
+but Teams is actually still running in the system tray, or the "notification
+area".
+
+![The Teams tray icon](tray-icon-000.png)
+
+_fig 1: an example of the Teams tray icon in the notification area_.
+
+When users want to be able to "minimize to the tray", they want:
+* The window to no longer appear on the taskbar
+* The window to no longer appear in the alt-tab order
+
+When minimized to the tray, it's almost as if there's no window for the Terminal
+at all. This can be combined with the global hotkey to quickly restore the window.
+
+The tray icon could be used for a variety of purposes. As a simple start, we
+could include the following three options:
+
+```
+Focus Terminal

```

Comment:
```suggestion
Focus Terminal
---
Summon Quake window
```
We should add the quake window here too

Reply:
I mean, that'll just be in the list of windows, under the name `_quake`, so that feels redundant

Reply:
Not necessarily. If you don't have a window that is "_quake", it won't be there.

And if you have a lot of windows open, that'll be a pain to find and click.

Adding this option makes it so you don't even need a global hotkey to summon the quake window.

Reply:
(Adding this to future considerations)